plugin/core/views.py

```diff
@@ -307,39 +337,38 @@ def formatting_options(settings: sublime.Settings) -> Dict[str, Any]:
     }
 
 
-def text_document_formatting(view: sublime.View) -> Request:
+def text_document_formatting(config: ClientConfig, view: sublime.View) -> Request:
```

Comment:
Just seeing all those similar changes makes me want to look into ways to refactor this.

Could create some kind of abstraction object (maybe `TextDocument` to match the spec) that would include methods to get `text_document_identifier` (which is what those two arguments seem to be needed for typically) and the path mapping function.

Could be initialized with `filename_or_view` and the `config` (or even `session`).

Just a wild idea...

Reply:
Would that work for the stuff in plugin/references.py (for example) as well?

Reply:
I have no idea if it would work in general ;)